### DISCUSSION TITLE
Poor image rendering on iOS - gpu mode

### DISCUSSION THREAD

**Discussion by MalacTheLittle** [airsdk/Adobe-Runtime-Support#2363] [Q&A]
Is there a way to get better image rendering on iOS in gpu mode? Same image on Android renders as it should...
JPG size is 3285 x 6000 px... 
(focus on the white text)

iOS screenshot: 
![iOS](https://user-images.githubusercontent.com/56675407/208273393-eee2c2f2-8189-493e-ad2d-719e7e3abb9c.PNG)

Android screenshot:
![android](https://user-images.githubusercontent.com/56675407/208273474-e12e69fa-89fc-4904-9d48-c279f4bfde54.jpg)

**Comment by MalacTheLittle** [comment]
@ajwfrost Is this a bug or am I doing something wrong? Please help!

**Reply by ajwfrost** [reply]
Hi - do you know what resolution it's being scaled down to? and how are you doing the scaling, is this a JPG loaded into a Loader object and then added onto the stage or are you getting the JPG into a BitmapData object and then drawing it somehow?

Would also be interested to know if the display changes if you're using different renderMode values on iOS..?

thanks

**Reply by MalacTheLittle** [reply]
Hi! 
Yes, I was testing it on iPad Air (2048×1536), iPhone 5S (640×1136) and iPhone 11 (1792×828), all the same results. 
I'm loading JPG file with `Loader.loadBytes()`, creating bitmap from loaded content and adding it in MovieClip that is resized to screensize.
CPU and Direct renderMode renders it fine, but app performance is degraded because of it ofcourse.

I was testing different image sizes and found that somewhere from 4100 px width it starts to degrade. But if I split image in two files of 3285×3000 and place 'em in same MovieClip rendering is fine.

If you need I can send you .ipa or .apk file.

Thanks

**Reply by MalacTheLittle** [reply]
I've also tried with different StageQuality settings, but no success.

**Reply by ajwfrost** [reply]
If you are able to send us an IPA (or just a sample app in ActionScript) we can have a look, as there are a few different ways to do the "creating bitmap from loaded content" bit as well as the resizing..
Upload link: https://transfer.harman.com/requests/aJVd5vQfdyMlRtYkm347zH

thanks

**Reply by MalacTheLittle** [reply]
@ajwfrost Do you have any updates on this issue? Problem still exists in newest AIR SDK.

**Reply by ajwfrost** [reply]
Hi

I just looked back at this one and had a little play with some test cases .. we see some poor scaling results in general when using a very large JPEG, but nothing that's particularly unexpected and we're not seeing any difference when using different render modes .. you had mentioned that it looked okay with cpu/direct modes and was just a problem with gpu mode?

Is it possible to get a test case of this, so that we can see exactly how the scaling is working and try to see if there's a configuration or setting we can change here?

thanks
